sample_keyprovider/README.md

```diff
@@ -1,20 +1,61 @@
-# README
+# CoCo Keyprovider
 
-Sample Keyprovider is a very simple Keyprovider skeleton, which currently has the following functions:
+CoCo Keyprovider is a very simple keyprovider tool, which can help to generate CoCo-compatible encrypted images.
+The encrypted image can be decrypted using `cc-kbc`, `offline-fs-kbc`, `offline-sev-kbc`, `online-sev-kbc` and `eaa-kbc`.
 
-1. It can be used to encrypt the container image to test the KBC module in AA. Sample Keyprovider provides a modular encryption mechanism. Different encryption modules can be used according to different kbcs to be tested.
+Also, historical toy KBC `sample kbc` is also supported.
 
-Sample Keyprovider can be a reference code base for you to quickly build your own KBS.
+The following guide will help make an encrypted image using CoCo keyprovider and decrypt that.
 
-## Test KBC modules
+## Related tools
 
-### Encryption
+Let's firstly introduce the related tools to help generate keys.
 
-Build and run sample keyprovider. You must explicitly specify the encryption module to use through the --features parameter (sample_enc for example): 
+### Test Key Generation and Offline-Fs-KBC
+
+`tools/generate_keys.sh` is a helper script to generate ten different base64-encoded KEK. Also, it can help to generate the `aa-offline_fs_kbc-keys.json` used by offline-fs-kbc.
+**Warning**: `jq` is required, so please install jq before use the script. For example on ubuntu, `apt install jq` could help.
+
+#### Generate `aa-offline_fs_kbc-keys.json`
+
+```bash
+tools/generate_keys.sh generate > aa-offline_fs_kbc-keys.json
+```
+
+you will get a randomly generated `aa-offline_fs_kbc-keys.json` like the following
+```
+{
+  "test-repo/key/key_id7": "gH4b2MYb/lGrYRVmrf3qtNo46mzC6C87cN5hdRt+Wvg=",
+  "test-repo/key/key_id6": "cemWcxJw9fG11Y3WhrzcZNKBWebdFt7qQDsIa6h6kJI=",
+  "test-repo/key/key_id10": "FePQCdgX/UONTIOl6eQP+7Itq5fdwr8JPufkmypbMWI=",
+  "test-repo/key/key_id5": "uDAikSR/Hadk7RXkA86DBsQs8eekP/dkLkXyuVjyLf8=",
+  "test-repo/key/key_id4": "5PN33vLpX+UO82Ryl/BhZNm5+suhTrBEqdWTDi1wqVU=",
+  "test-repo/key/key_id3": "Y8ptYp5WgJxXdhe5tIS+ZIoSlVup1DUHu63MyHOsxsA=",
+  "test-repo/key/key_id2": "sIDb9K+J7IEfYQMbcDzEYz8t+hABQlHSR+55SIAgsUw=",
+  "test-repo/key/key_id1": "TfyYGL/gBbtXwgDmx3a6N6WxFJFamcSRUFlpBPXu0f4=",
+  "test-repo/key/key_id9": "CfGZPsmKq1pkzraBpMAsbBZGIGlbWUepu4eR/Z4exnE=",
+  "test-repo/key/key_id8": "eXIlv83nTjfyeLZcCvTda9ypYIYj83eGjbqjoVFAQHA="
+}
+```
+
+#### Use the Generated Key to Encrypt Image
+
+If we want to use key of id `test-repo/key/key_id1` in `aa-offline_fs_kbc-keys.json` to encrypt an image, we firstly export the key content to file `key1`
+
+```
+tools/generate_keys.sh export aa-offline_fs_kbc-keys.json test-repo/key/key_id1 key1
```

Comment:
Can we not pass `aa-offline_fs_kbc-keys.json`, but directly use the tool to generate a key?

Reply:
We can generate a random key using `/dev/random` now. As the feature, I think when resource registration is prepared in KBS, we can integrate this function together with kbs, s.t.
- randomly generate a key
- use the key to encrypt image
- register the key into kbs

Reply:
If not in hurry, we can impl that in another PR when all basic functions are ok

Reply:
Yeah, I agree we should try to create a seamless flow for encrypted and provisioning images.